lib/web_ui/lib/src/engine/canvas_pool.dart

```diff
@@ -443,7 +448,11 @@ class _CanvasPool extends _SaveStackTracking {
     if (_canvas != null) {
       html.CanvasRenderingContext2D ctx = context;
       _runPath(ctx, path as SurfacePath);
-      ctx.clip();
+      if (path.fillType == ui.PathFillType.nonZero) {
+        ctx.clip();
+      } else {
+        ctx.clip('evenodd');
+      }
```

Comment:
I'm curious why you did the opposite check here compared to the above. Should we keep them consistent?

Reply:
Done.